danskernesdigitalebibliotek/dpl-cms
# Add publication date to articles and pages - Ddfform 639

#### Link to issue
https://reload.atlassian.net/browse/DDFFORM-639

#### Description
This pull request introduces the `dpl_publication` module, which manages the publication dates for `article`s and `page`s. It automatically sets the publication date when content is published and includes a deploy hook that updates the `field_publication_date` with the value "created time" for each node if not already set.

Additionally, it adds the `field_publication_date` field to the `article` and `page` templates and sorts the views by the `field_publication_date`.

I have manually updated the default content since the `default-content` module executes after `dpl_publication` the `field_publication_date` field is not updated by the `dpl_publication_deploy_default_publication_date` hook.

Here is a video that demonstrates that the deploy hook works if there is data when the deploy hook runs

https://github.com/danskernesdigitalebibliotek/dpl-cms/assets/49920322/325273c8-688f-4025-8637-29bc7f6f76bf



#### How to test
Try un publishing an article manually or via a schedule (remember to run the cron job) and see the publish date change reflected in the templates.

https://varnish.pr-1250.dpl-cms.dplplat01.dpl.reload.dk/node/12/edit


https://varnish.pr-1250.dpl-cms.dplplat01.dpl.reload.dk/articles


Templates
